.github/workflows/release.yml

```diff
@@ -0,0 +1,48 @@
+name: Publish to PyPI
+
+on:
+  push:
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'
+
+jobs:
+  build-test:
+    uses: ./.github/workflows/build-test.yml
+
+  publish:
+    needs: [build-test]
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest ]
+        python-version: [ "3.9" ]
+    runs-on: ${{ matrix.os }}
+    permissions:
+      id-token: write
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip setuptools wheel
+          python -m pip install 'sphinx-theme-builder[cli]'
+
+      - name: Build theme
+        run: stb package
+
+      - name: Publish to PyPI
```

Comment:
We will want to create a custom build for this part and also an environment name for PyPI trusted workflows - see what we did here: 

https://github.com/pyOpenSci/pyosMeta/blob/main/.github/workflows/publish-pypi.yml#L50